lib/src/layout/resizable_pane.dart

```diff
@@ -0,0 +1,180 @@
+  final bool? isResizable;
+
+  /// Specifies the maximum width that this [ResizablePane] can have.
+  ///
+  /// The value can be null and defaults to `500.0`.
+  final double? maxWidth;
+
+  /// Specifies the minimum width that this [ResizablePane] can have.
+  final double minWidth;
+
+  /// Indicates the draggable side of the sidebar for resizing
+  final ResizableSide resizableSide;
+
+  /// Specifies the width of the scaffold at which this [ResizablePane] will be hidden.
+  final double? scaffoldBreakpoint;
+
+  @override
+  ResizablePaneState createState() => ResizablePaneState();
+}
+
+class ResizablePaneState extends State<ResizablePane> {
+  final _key = UniqueKey();
+  final _scrollController = ScrollController();
+  late double _width;
+
+  Color? get _dividerColor => context.macosTheme.dividerColor;
+
+  ScaffoldScope get _scaffoldScope => ScaffoldScope.of(context);
+
+  ResizablePaneNotifier get _notifier => _scaffoldScope.valueNotifier;
+
+  BoxConstraints get _constraints => _scaffoldScope.constraints;
+
+  double? get _maxWidth => _constraints.maxWidth;
+
+  double? get _maxHeight => _constraints.maxHeight;
+
+  bool get _resizeOnRight => widget.resizableSide == ResizableSide.right;
+
+  BoxDecoration get _decoration {
+    final _borderSide = BorderSide(color: _dividerColor!);
+    final right = Border(right: _borderSide);
+    final left = Border(left: _borderSide);
+    return BoxDecoration(border: _resizeOnRight ? right : left).copyWith(
+      color: widget.decoration?.color,
+      border: widget.decoration?.border,
+      borderRadius: widget.decoration?.borderRadius,
+      boxShadow: widget.decoration?.boxShadow,
+      backgroundBlendMode: widget.decoration?.backgroundBlendMode,
+      gradient: widget.decoration?.gradient,
+      image: widget.decoration?.image,
+      shape: widget.decoration?.shape,
+    );
+  }
+
+  Widget get _resizeArea {
+    return GestureDetector(
+      behavior: HitTestBehavior.opaque,
+      child: MouseRegion(
+        cursor: SystemMouseCursors.resizeColumn,
+        child: SizedBox(width: 5),
+      ),
+      onHorizontalDragUpdate: (details) {
+        setState(() {
+          _width = math.max(
+            widget.minWidth,
+            math.min(
+              math.min(widget.maxWidth ?? 0, _maxWidth!),
+              _resizeOnRight
+                  ? _width + details.delta.dx
+                  : _width - details.delta.dx,
+            ),
+          );
+          if (_width >= widget.minWidth && _width < widget.maxWidth!) {
+            _notifier.update(_key, _width);
+          }
+        });
+      },
+    );
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    _width = widget.minWidth;
+    _scrollController.addListener(() => setState(() {}));
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(widget.maxWidth! >= widget.minWidth);
```

Comment:
If you're gonna keep it this way, make sure to check if `widget.maxWidth` is not null instead of adding `!`.

```dart
if (widget.maxWidth != null)
  assert(widget.maxWidth >= widget.minWidth)
```

This can also be checked on the widget constructor (I recommend that way):

```dart
  const ResizablePane({
    Key? key,
    required this.builder,
    this.decoration,
    this.maxWidth = 500.0,
    required this.minWidth,
    this.isResizable = true,
    required this.resizableSide,
    this.scaffoldBreakpoint = 500,
  }) : assert(maxWidth == null || maxWidth >= minWidth), super(key: key);
```